### ISSUE TITLE
[client] Accessibility: Announce when I delete annotations

### ISSUE THREAD

**Issue by chrisshaw** [hypothesis/product-backlog#1343]
**As a user using assistive technologies like NVDA, JAWS, or VoiceOver, I want to be notified when I delete an annotation, so that I can move on confidently to engage in discussions more quickly.**

A reviewer in an accessibility review considered it a major defect that, upon deleting an annotation, the user would not know anything actually happened:

> No confirmation that the deletion was successful
> Speak a confirmation message.
> Major

This fits a theme of missing announcements as users using assistive technologies interact with the app, and it completes the delete flow covered in #1342 .

## Related

Related to #1342

**Comment by chrisshaw** [comment]
**Issue was linked to Harvestr Discovery:**  [Hear announcements when I delete annotations via screen readers](https://app.harvestr.io/components/0/list/I4lgxgB-f)

**Comment by lyzadanger** [comment]
I _believe_ we may be able to address this by extending our current "toast messages" UI to support visually-hidden toast messages. In that approach, a sighted user would not see a toast message on deleting an annotation but a visually-hidden toast message would be added and screen readers should pick that up.

I've read that elements with `display: none` do not always work with `aria-live`(which is used by our Toast Messages) and screenreaders, but I'm holding out hope that using a visually-hidden/"screen-reader-only" approach _will_.

In the eventuality that we have trouble getting screen-readers to announce visually-hidden toast messages, we have the option of displaying a visual confirmation on annotation delete (i.e. all users would see it) if that's acceptable.

Which begs the question: to confirm, we _don't_ want to show a toast message for everyone when an annotation is deleted...do we? Because that would be even more dead simple.

**Comment by chrisshaw** [comment]
@lyzadanger Confirming we don't want to add status messages for this.

<details>
<summary>How I was thinking about it</summary>

I do think deletion status messages could be useful, but they become more valuable when there is an action that can be taken on that status message (e.g., undo). I also got to thinking that, should we want to introduce undo, we could likely do better than an undo button in the toast message (which I have always though a poor UX: "You accidentally deleted something; if you want it back, you have 5 seconds to find the message, land your cursor/finger in the right tap target and click before everything self destructs. The clock already started!")

We also definitely don't want visual confirmation messages for the other saves, so we'd still have to do the extension work you spoke about.

</details>
